src/components/utils/src/timer.cc

```diff
@@ -64,6 +64,8 @@ timer::Timer::~Timer() {
   StopDelegate();
   single_shot_ = true;
 
+  thread_->set_delegate(NULL);
```

Comment:
@anosach-luxoft I am aware of the `ThreadDelegate` destructor behavior however it is unintuitive and is not documented. There is no clear indication that it will stand the same so I decided to take precautions.